scripts/crd-tools/pkg/objectvisitor/objectvisitor.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectvisitor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	kccyaml "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+)
+
+type Visitor interface {
+	VisitObject(obj *unstructured.Unstructured) error
+}
+
+func VisitObjectsInDirectory(ctx context.Context, dir string, visitor Visitor) error {
+	if err := filepath.WalkDir(dir, func(p string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		if filepath.Ext(p) != ".yaml" {
+			return nil
+		}
```

Comment:
this is a bikeshedding thought but we probably want to avoid files like `kustomize.yaml` (and all of its forms) if we use a mix of our own tooling and controller gen.